src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDecreaseInheritedMemberVisibilityTests.cs

```diff
@@ -12,12 +10,295 @@ public class DoNotDecreaseInheritedMemberVisibilityTests : DiagnosticAnalyzerTes
+    public void MyProp3() {}
+}
+
+public class DerivedClass : BaseClass
+{
+    private void MyProp3() {}
+}", GetCSharpCA2222RuleNameResultAt(8, 18));
+        }
+
+        [Fact]
+        public void DecreaseBasicMemberVisibility()
+        {
+            VerifyBasic(
+@"Public Class BaseClass
+    Public MyProp3 As Integer
+End Class
+
+Public Class DerivedClass
+    Inherits BaseClass
+    Private Shadows MyProp3 As Integer
+End Class", GetBasicCA2222RuleNameResultAt(7, 21));
+
+            VerifyBasic(
+@"Public Class BaseClass
+    Public ReadOnly Property MyProp3() As Integer
+End Class
+
+Public Class DerivedClass
+    Inherits BaseClass
+    Friend Shadows ReadOnly Property MyProp3() As Integer
+End Class", GetBasicCA2222RuleNameResultAt(7, 38));
+
+            VerifyBasic(
+@"Public Class BaseClass
+    Public Property MyProp3() As Integer
+        Get
+            Return m_MyProp3
+        End Get
+        Set
+            m_MyProp3 = Value
+        End Set
+    End Property
+    Private m_MyProp3 As Integer
+End Class
+
+Public Class DerivedClass
+    Inherits BaseClass
+    Public Shadows Property MyProp3() As Integer
+        Get
+            Return m_MyProp3
+        End Get
+        Private Set
+            m_MyProp3 = Value
+        End Set
+    End Property
+    Private Shadows m_MyProp3 As Integer
+End Class", GetBasicCA2222RuleNameResultAt(19, 17));
+
+            VerifyBasic(
+@"Public Class BaseClass
+    Public Sub MyProp3()
+    End Sub
+End Class
+
+Public Class DerivedClass
+    Inherits BaseClass
+    Private Overloads Sub MyProp3()
+    End Sub
+End Class", GetBasicCA2222RuleNameResultAt(8, 27));
+        }
+
+        [Fact]
+        public void IncreaseCSharpMemberVisibility()
+        {
+            VerifyCSharp(
+@"public class BaseClass
+{
+    internal int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3;
+}");
+        }
+
+        [Fact]
+        public void IncreaseBasicMemberVisibility()
+        {
+            VerifyBasic(
+@"Public Class BaseClass
+    Friend MyProp3 As Integer
+End Class
+
+Public Class DerivedClass
+    Inherits BaseClass
+    Public Shadows MyProp3 As Integer
+End Class");
+        }
+
+        [Fact]
+        public void SimilarCSharpVisibilities()
+        {
+            VerifyCSharp(
+@"public class BaseClass
+{
+    internal int MyProp3;
+}
+
+public class DerivedClass : BaseClass
+{
+    private new int MyProp3;
+}");
+
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public int MyProp3 {get;}
+}
+
+public class DerivedClass : BaseClass
+{
+    public new int MyProp3 {get;}
+}");
+
+            VerifyCSharp(
+@"public class BaseClass
+{
+    internal void MyProp3() {}
+}
+
+public class DerivedClass : BaseClass
+{
+    protected void MyProp3() {}
+}");
+
+            VerifyCSharp(
+@"public class BaseClass
+{
+    public void MyProp3() {}
+}
+
+public class DerivedClass : BaseClass
+{
+    protected void MyProp3() {}
+}");
+        }
```

Comment:
The docs are a little misleading. The way the rule really works is to look for instances of changing from 'visible outside the assembly' (public, protected and not sealed) to not 'visible outside the assembly'. So, dropping from public to protected doesn't get flagged.